specification/cognitiveservices/OpenAI.Inference/models/completions/extensions/oyd/common_options.tsp

```diff
@@ -38,4 +38,40 @@ alias OnYourDataCommonChatExtensionConfiguration = {
   @doc("Give the model instructions about how it should behave and any context it should reference when generating a response. You can describe the assistant's personality and tell it how to format responses. There's a 100 token limit for it, and it counts against the overall token limit.")
   @encodedName("application/json", "role_information")
   roleInformation?: string;
+  
+  /** 
+   * The max number of rewritten queries should be send to search provider for one user message. If not specified, 
+   * the system will decide the number of queries to send. 
+   */
+  @encodedName("application/json", "max_search_queries")
+  @added(ServiceApiVersions.v2024_05_01_Preview)
+  maxSearchQueries?: int32;
+
+  /** 
+   * If specified as true, the system will allow partial search results to be used and the request fails if all the queries fail. 
+   * If not specified, or specified as false, the request will fail if any search query fails.
+   */
+  @encodedName("application/json", "allow_partial_results")
```

Comment:
I believe this is `allow_partial_result` (no trailing 's'):
https://github.com/Azure/azure-rest-api-specs/blob/cb1185d9961b7dabe002fdb4c3a28c07d130e47e/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2024-05-01-preview/inference.yaml#L2697

(I like your name better, but the *I suppose* the encoding needs to match)